src/lib/y2security/autoinst/lsm_config_reader.rb

```diff
@@ -18,55 +18,47 @@
 # find current contact information at www.suse.com.
 
 require "y2security/lsm/config"
-require "y2security/autoinst_profile"
+require "y2security/autoinst_profile/security_section"
 
 module Y2Security
   module Autoinst
     # This class is responsible of reading the Linux Security Module configuration declared in
     # the AutoYaST profile
     class LSMConfigReader
-      # @return [AutoinstProfile::LSMSection]
+      # @return [AutoinstProfile::SecuritySection]
       attr_reader :section
-      # @return [AutoinstProfile::SelinuxSection, AutoinstProfile::ApparmorSection, nil]
-      attr_reader :module_section
-
       # Constructor
       #
-      # @param section [AutoinstProfile::LSMSection]
+      # @param section [AutoinstProfile::SecuritySection]
       def initialize(section)
         @section = section
       end
 
       # Reads the Linux Security Module configuration defined in the profile modifying it
       # accordingly
       def read
-        return unless section
+        return unless section.lsm_select || section.selinux_mode
 
-        config.configurable = section.configurable
-        config.select(section.select) if section.select
-        configure_supported_modules
+        select_module
+        configure_selinux if selinux?
       end
 
     private
 
-      def configure_supported_modules
-        [:selinux, :apparmor, :none].each do |id|
-          lsm_module = config.public_send(id)
-          @module_section = section.public_send(id)
-          next unless module_section
+      def selinux?
+        return true if section.lsm_select == "selinux"
+        return true if !section.lsm_select && section.selinux_mode
 
-          assign(lsm_module, :mode) if id == :selinux
-          assign(lsm_module, :selectable)
-          next if id == :none
+        false
+      end
 
-          assign(lsm_module, :configurable)
-          assign(lsm_module, :patterns)
-        end
+      def configure_selinux
+        config.selinux.mode = section.selinux_mode
```

Comment:
what if section is missing mode. Is it ok to have it as nil?

Reply:
Yes it is ok. If there is no section then the defaults will be computed during the proposal which will try to select the default **LSM** based on the control file and the same happens with the SELinux mode if it is not set explicitly.

See https://github.com/yast/yast-security/blob/master/src/lib/y2security/lsm/selinux.rb#L121